### PR TITLE
Finished task 10 (make twitter dynamic)

### PR DESCRIPTION
**Pull request by Sisha3342**
Also placed two .txt files to UI folder with instructions on using localStorage and registeredUsers;
(task wasn't performed in project folder because the 8'th pull request is still not approved)

### CODE REVIEW

File: UI/js/PostsList.js [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+
+        this.saveToLocalStorage();
     }
 
-    getPage(skip=0, top=this._posts.length, filterConfig=undefined) {
+    getPage(skip=0, top=10, filterConfig=undefined) {
         let postsToReturn = this._posts.concat();
 
         if (filterConfig) {
             for(let property in filterConfig) {
```
Comment (MaksimViarenich) [inline UI/js/PostsList.js:12]: Minor. Check everywhere code style `for () {`

File: UI/js/PostsList.js [hunk 2]
```diff
@@ -137,6 +150,31 @@ class PostsList {
...
+    getLength() {
+        return this._posts.length;
+    }
+
+    saveToLocalStorage() {
+        localStorage.setItem('posts', JSON.stringify(this));
+
+        // for (let i = 0; i < this.getLength(); i++) {
+        //     localStorage.setItem('post' + i, JSON.stringify(this._posts[i]));
+        // }
```
Comment (MaksimViarenich) [inline UI/js/PostsList.js:163]: Remove commented code, please

File: UI/js/PostsList.js [hunk 3]
```diff
@@ -31,7 +44,7 @@ class PostsList {
             }
         });
```
Comment (MaksimViarenich) [inline UI/js/PostsList.js:45]:
```
postsToReturn.sort(function (post1, post2) {
    if (post1.createdAt < post2.createdAt) {
        return 1;
    }

    if (post2.createdAt < post1.createdAt) {
        return -1;
    }
 
    return 0;
});
```

File: UI/js/View.js [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        if (this._currentUser === '') {
+            document.querySelector('.add-button').style.visibility = 'hidden';
+        }
+    }
+
+    displayPage(page) {
+        let pageHandlers = {addPostPage: View._showAddPostPage,
+                            editPostPage: View._showEditPostPage,
+                            mainPage: View._showMainPage,
+                            authPage: View._showAuthPage};
```
Comment (MaksimViarenich) [inline UI/js/View.js:86]: You have one of the most beautiful code!
But better:
```
let pageHandlers = {
    addPostPage: View._showAddPostPage,
    editPostPage: View._showEditPostPage,
    mainPage: View._showMainPage,
    authPage: View._showAuthPage
};
```

File: UI/js/View.js [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+        <div class="news-seed">
+            <button class="add-button">Add post</button>
+
+            <div class="posts">
+            </div>
+            
+            <button class="more-button">Load more</button>
+        </div>
+        `
```
Comment (MaksimViarenich) [inline UI/js/View.js:144]: Where is ;?

File: UI/js/View.js [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+                <h2>Description</h2>
+                <textarea name="postDescriptionInput" rows="10"></textarea>
+                
+                <h2>Image</h2>
+                <input name="postImageInput" type="file" accept=".jpg, .jpeg, .png">
+            </form>
+            
+            <button class="add-post-button">Add post</button>
+            <button class="back-button">Back</button>
+        `
```
Comment (MaksimViarenich) [inline UI/js/View.js:183]: Add ;

File: UI/js/View.js [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+            <h2>Description</h2>
+            <textarea name="postDescriptionInput" rows="10"></textarea>
+            
+            <h2>Image</h2>
+            <input name="postImageInput" type="file" accept=".jpg, .jpeg, .png">
+        </form>
+
+        <button class="edit-post-button">Edit post</button>
+        <button class="back-button">Back</button>
+        `
```
Comment (MaksimViarenich) [inline UI/js/View.js:218]: Add ;

File: UI/js/View.js [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+            <h2>Username</h2>
+            <input name="userNameInput" type="text" placeholder="user">
+            
+            <h2>Password</h2>
+            <input name="userPasswordInput" type="password">
+        </form>
+
+        <button class="log-in-button">Log in</button>
+        <button class="back-button">Back</button>
+        `
```
Comment (MaksimViarenich) [inline UI/js/View.js:253]: Add;

File: UI/js/PostsList.js [hunk 9]
```diff
@@ -1,20 +1,33 @@
 class PostsList {
```
Comment (MaksimViarenich) [inline UI/js/PostsList.js:1]: `addAll, clear, restoreFromLocalStorage` methods are not used

File: UI/js/createTestPosts.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+testPosts = new PostsList([...generatePosts(40)]);
+// testPosts = PostsList.restoreFromLocalStorage();
```
Comment (MaksimViarenich) [inline UI/js/createTestPosts.js:2]: Remove commented code

**Comment by Sisha3342** [comment]
Sorry, but I can't open the video. Also *restoteLocalPage()* is needed for uploading posts if it's not the first attempt. So, when I upload my page for the first time, I create posts with **generator**.  Then I comment this line and uncomment the second to get those posts from **local storage**.

**Comment by Sisha3342** [comment]
Ended fixing code style and application logic

**Pull request merged**